spec/system/admin/products_v3/products_spec.rb

```diff
@@ -1336,6 +1336,38 @@
             end
           end
         end
+
+        context 'a shipped product' do
+          let!(:order) { create(:shipped_order, line_items_count: 1) }
+          let!(:line_item) { order.reload.line_items.first }
+
+          context "a deleted line item from a shipped order" do
+            before do
+              login_as_admin
+              visit admin_products_url
+
+              # Delete Variant
+              within variant_selector do
+                page.find(".vertical-ellipsis-menu").click
+                page.find(delete_option_selector).click
+              end
+
+              delete_button_selector = "input[type=button][value='Delete variant']"
+              within modal_selector do
+                page.find(delete_button_selector).click
+              end
+
+              expect(page).not_to have_selector(modal_selector)
+              expect(page).not_to have_selector(variant_selector)
```

Comment:
Personally I don't like having `expect` in a before block, I would assume that deleting the variant is covered somewhere else so the `expect` are not necessary. If not I would just move this to the `it` block, we only have on `it` in this `context` to I think it's fine to have the set up in the `it` block.

Reply:
Makes sense - indeed, we do have the Delete scenario covered in another section of this spec; no point in testing it again - removed these two (negative) assertions.